<launch>
<!-- Launch file for ouster OS0-128 LiDAR -->

    <arg name="rviz" default="false" />
    <param name="/use_sim_time" value="true" />

    <rosparam command="load" file="$(find fast_lio)/config/nce_ouster128.yaml" />

    <arg name="num_filter_points"/>
    <arg name="size_surf"/>
    <arg name="size_map"/>

    <param name="dont_compensate" type="bool" value="false"/>
    <param name="has_lidar_end_time" type="bool" value="false"/>
    <param name="feature_extract_enable" type="bool" value="false"/>
    
    <param name="max_iteration" type="int" value="3" />
    <param name="cube_side_length" type="double" value="1000" />
    <param name="runtime_pos_log_enable" type="bool" value="false" />
    <node  pkg="fast_lio" type="compensate_intensity" name="compensate_intensity" output="screen"/> 
    <node  pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" required="true"/> 
    <!-- node launch-prefix="gdb -ex run -args " pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" / --> 

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/loam_livox.rviz" />
    </group>

</launch>
